modules/CompanyUser/Repositories/CompanyUserRepository.php

```
@@ -548,12 +548,12 @@ private function handleOwnerPermissions(User $user, $companyId): void
 
             $branch->update(["manager_id" => $user->id]);
 
-            $this->managementHierarchyRepository->findOneBy([
+            $this->managementHierarchyRepository->model->where([
                 "company_id" => $companyId,
                 "parent_id" => $branch->id,
                 "type" => "management",
                 "is_main" => 1
-            ])->update(["manager_id" => $user->id]);
+            ])->first()->update(["manager_id" => $user->id]);
```

The change from `findOneBy` to `where()->first()` might introduce a potential issue if no record is found. The original code would silently fail if no record exists, while the new code would throw an exception when trying to call `update()` on null. Consider adding a null check before calling `update()`.